sdk/state/update_test.go

```diff
@@ -80,3 +82,63 @@ func TestLastConfirmedPayment(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, Payment{}, receiverChannel.latestUnconfirmedPayment)
 }
+
+func TestAppendNewSignature(t *testing.T) {
+	localSigner := keypair.MustRandom()
+	remoteSigner := keypair.MustRandom()
+	localEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(101),
+	}
+	remoteEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(202),
+	}
+	sendingChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		Initiator:           true,
+		LocalSigner:         localSigner,
+		RemoteSigner:        remoteSigner.FromAddress(),
+		LocalEscrowAccount:  localEscrowAccount,
+		RemoteEscrowAccount: remoteEscrowAccount,
+	})
+
+	rand1, err := randomByteArray(10)
+	require.NoError(t, err)
+	rand2, err := randomByteArray(10)
+	require.NoError(t, err)
+	newCloseSignatures := []xdr.DecoratedSignature{
+		xdr.DecoratedSignature{
+			Signature: rand1,
+		},
+		xdr.DecoratedSignature{
+			Signature: rand2,
+		},
+	}
+
+	sendingChannel.latestUnconfirmedPayment.CloseSignatures = appendNewSignatures(sendingChannel.latestUnconfirmedPayment.CloseSignatures, newCloseSignatures)
+	assert.Equal(t, 2, len(sendingChannel.latestUnconfirmedPayment.CloseSignatures))
+	assert.Equal(t, newCloseSignatures[0], sendingChannel.latestUnconfirmedPayment.CloseSignatures[0])
+	assert.Equal(t, newCloseSignatures[1], sendingChannel.latestUnconfirmedPayment.CloseSignatures[1])
+
+	// 1 new signature is introduced
+	rand3, err := randomByteArray(10)
+	require.NoError(t, err)
+	newCloseSignatures = append(newCloseSignatures, xdr.DecoratedSignature{
+		Signature: rand3,
+	})
+	sendingChannel.latestUnconfirmedPayment.CloseSignatures = appendNewSignatures(sendingChannel.latestUnconfirmedPayment.CloseSignatures, newCloseSignatures)
+	assert.Equal(t, 3, len(sendingChannel.latestUnconfirmedPayment.CloseSignatures))
+	for i, ncs := range newCloseSignatures {
+		assert.Equal(t, ncs, sendingChannel.latestUnconfirmedPayment.CloseSignatures[i])
+	}
+}
+
+func randomByteArray(length int) ([]byte, error) {
+	arr := make([]byte, length)
+	_, err := rand.Read(arr)
+	if err != nil {
+		return nil, err
+	}
+	return arr, nil
+}
```

Comment:
🤔 It's not clear to me what this test is testing. There's a lot of setup relating to channels, but then the functions that are called are just `appendNewSignatures` on groups of signatures. If this function is only intended to test `appendNewSignatures` then you don't need any of the `Channel` setup.

Would the below achieve the same thing you're going for?

```suggestion
func TestAppendNewSignature(t *testing.T) {
	closeSignatures := []xdr.DecoratedSignature{
		{ Signature: randomByteArray(10) },
		{ Signature: randomByteArray(10) },
	}
	
	closeSignaturesToAppend := []xdr.DecoratedSignature{
		closeSignatures[0], // A duplicate signature is included.
		{ Signature: randomByteArray(10) },
	}

	newCloseSignatures := appendNewSignatures(closeSignatures, closeSignaturesToAppend)
	
	// Check that the final slice of signatures does not contain the duplicate.
	assert.ElementsMatch(
		t,
		newCloseSignatures,
		[]xdr.DecoratedSignature{
			closeSignatures[0],
			closeSignatures[1],
			closeSignaturesToAppend[1],
		},
	)
}

func randomByteArray(t *testing.T, length int) []byte {
	arr := make([]byte, length)
	_, err := rand.Read(arr)
	require.NoError(t, err)
	return arr
}
```

Reply:
ya  I think I was too tied to using the channel instead of focusing on the append method. This is a great simplification. Also `assert.ElementsMatch` 🙌 
https://github.com/stellar/experimental-payment-channels/pull/88/commits/39ddf70ab299af6f2f69e5e64739be0915a88405